### PR TITLE
Fix compatibility issues with xray mutation and vision-modifying items

### PR DESCRIPTION
**Pull request by PrimeNumb**
## About the PR 
- Xray mutation is now compatible with mesons, thermals, ecto goggles, etc
- Parts of sight code refactored to use mob properties

## Why's this needed? 
fixes https://github.com/goonstation/goonstation/issues/5529
cleans up sight code ever so slightly

### CODE REVIEW

**Comment by alexeric123** [comment]
Does this also work with other interactions other than the x-ray mutation?
Such as night vision mutation+mesons/thermals
or night vision mutation + xray mutation

**Comment by Tarmunora** [comment]
Might be worth adding handling to the sight properties to run the sight process when the property changes

**Comment by PrimeNumb** [comment]
> Does this also work with other interactions other than the x-ray mutation?
> Such as night vision mutation+mesons/thermals
> or night vision mutation + xray mutation

Correct.

> Might be worth adding handling to the sight properties to run the sight process when the property changes

Done.

File: _std/macros/mob_properties.dm [hunk 1]
```diff
@@ -161,6 +161,11 @@ To remove:
 	SEND_SIGNAL(target, COMSIG_MOB_PROP_INVISIBILITY, old_val); \
 	} while(0)
 
+#define PROP_UPDATE_SIGHT(target, prop, old_val) do {\
+	if(!istype(target, /mob/living)) return; \
+	var/mob/living/_M = target; \
+	_M.lifeprocesses[/datum/lifeprocess/sight].process(); \
```
Comment (PrimeNumb) [inline _std/macros/mob_properties.dm:167]: tmk there should be no instance where this lifeprocess does not exist (outside of disposing)

File: _std/macros/mob_properties.dm [hunk 2]
```diff
@@ -161,6 +161,14 @@ To remove:
 	SEND_SIGNAL(target, COMSIG_MOB_PROP_INVISIBILITY, old_val); \
 	} while(0)
 
+#define PROP_UPDATE_SIGHT(target, prop, old_val) do {\
+	if(!istype(target, /mob/living)) return; \
+	var/mob/living/_living_mob = target; \
+	var/datum/lifeprocess/sight/_sightprocess = _living_mob.lifeprocesses[/datum/lifeprocess/sight]; \
+	var/datum/lifeprocess/hud/_hudprocess = _living_mob.lifeprocesses[/datum/lifeprocess/hud]; \
+	_sightprocess.process(); \
+	_hudprocess.process(); \
```
Comment (PrimeNumb) [inline _std/macros/mob_properties.dm:170]: hud process is for tinting
not sure how expensive this gets for large amounts of players

File: code/obj/item.dm [hunk 3]
```diff
@@ -984,6 +984,12 @@
 		equipment_proxy.additive_slowdown -= spacemove
 		equipment_proxy.space_movement -= spacemove
 
+/obj/item/proc/after_unequip(var/mob/user)
```
Comment (PrimeNumb) [inline code/obj/item.dm:987]: `unequipped()` has this quirk where the item is technically still in the slot during the proc call.
I suspect a _shitton_ of things rely on this behavior, so I added `after_unequip()`. 
See `code/mob/living/carbon/human.dm` and `code/datums/hud/human.dm` on how it's called.

File: _std/macros/mob_properties.dm [hunk 4]
```diff
@@ -161,6 +161,11 @@ To remove:
 	SEND_SIGNAL(target, COMSIG_MOB_PROP_INVISIBILITY, old_val); \
 	} while(0)
 
+#define PROP_UPDATE_SIGHT(target, prop, old_val) do {\
+	if(!istype(target, /mob/living)) return; \
```
Comment (Tarmunora) [inline _std/macros/mob_properties.dm:165]:
```suggestion
	if(!isliving(target)) return; \
```

File: _std/macros/mob_properties.dm (hunk 1 above)
Comment (Tarmunora) [inline _std/macros/mob_properties.dm]:
```suggestion
	_M.lifeprocesses?[/datum/lifeprocess/sight]?.process(); \
```

**Review by Tarmunora — CHANGES_REQUESTED** [review]

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**